toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/TableTopSceneView.kt

```diff
@@ -102,62 +122,150 @@ fun TableTopSceneView(
     onDrawStatusChanged: ((DrawStatus) -> Unit)? = null,
     content: (@Composable TableTopSceneViewScope.() -> Unit)? = null
 ) {
-    val lifecycleOwner = LocalLifecycleOwner.current
+    Box(modifier = modifier) {
+        var initializationStatus: TableTopSceneViewInitializationStatus by remember {
+            mutableStateOf(
+                TableTopSceneViewInitializationStatus.Initializing
+            )
+        }
+        onInitializationStatusChanged?.invoke(initializationStatus)
+        val lifecycleOwner = LocalLifecycleOwner.current
+        val context = LocalContext.current
+        val cameraPermissionGranted by rememberCameraPermission(requestCameraPermissionAutomatically) {
+            // onNotGranted
+            initializationStatus = TableTopSceneViewInitializationStatus.FailedToInitialize(
+                IllegalStateException(context.getString(R.string.camera_permission_not_granted))
+            )
+            onInitializationStatusChanged?.invoke(initializationStatus)
+        }
 
-    val context = LocalContext.current
-    val arSessionWrapper = remember { ArSessionWrapper(context.applicationContext) }
-    DisposableEffect(Unit) {
-        lifecycleOwner.lifecycle.addObserver(arSessionWrapper)
-        onDispose {
-            lifecycleOwner.lifecycle.removeObserver(arSessionWrapper)
-            arSessionWrapper.onDestroy(lifecycleOwner)
+        var arCoreInstalled by remember { mutableStateOf(false) }
+
+        if (cameraPermissionGranted) {
+            // invoked when the cameraPermissionGranted state changes to true
+            LaunchedEffect(Unit) {
+                val arCoreAvailability = checkArCoreAvailability(context)
+                val isArCoreAvailable =
+                    arCoreAvailability == ArCoreApk.Availability.SUPPORTED_INSTALLED
+                if (!isArCoreAvailable) {
```

Comment:
Seems like `isArCoreAvailable` is redundant, instead you could...
```suggestion
                if (arCoreAvailability != ArCoreApk.Availability.SUPPORTED_INSTALLED) {
```